content/blog/2024-07-31-valkey-8-0-0-rc1.md

```diff
@@ -0,0 +1,126 @@
+  repairs the logical links between source and target nodes. This ensures requests
+  are correctly routed to the new primary in the target shard, maintaining cluster
+  integrity and availability.
+- **Seamless Support for New Replicas**: New replicas joining a shard mid-migration
+  will automatically inherit the correct slot migration state, ensuring immediate
+  consistency.
+
+## Replication
+
+Valkey 8.0 introduces a dual-channel replication scheme, allowing the RDB and
+the replica backlog to be transferred simultaneously, accelerating synchronization.
+
+- **Reduced Memory Load**: By streaming replication data to the replica during
+  the full sync, the primary node experiences significantly less memory pressure.
+  The replica now manages the Client Output Buffer (COB) tracking, reducing the
+  likelihood of COB overruns and enabling larger COB sizes on the replica side.
+- **Lighter Main Process Load**: A dedicated connection for RDB transfer frees
+  the primary's main process from handling this data, allowing it to focus on
+  client queries and improving overall responsiveness.
+
+Performance tests show improvements in write latency during sync, and in scenarios
+with heavy read commands, the sync time can be cut by up to 50%. This translates
+to a more responsive system, even during synchronization.
+
+## Observability
+
+Valkey 8.0 introduces a comprehensive per-slot metrics infrastructure, providing
+detailed visibility into the performance and resource usage of individual slots.
+This granular data helps inform decisions about resource allocation, load
+balancing, and performance optimization.
+
+- **CPU Usage**: Tracks CPU time consumed by operations on each slot, identifying
+  areas of high utilization and potential bottlenecks.
+- **Network Input/Output Bytes**: Monitors data transmission and reception by
+  each slot, offering insights into network load and bandwidth utilization.
+- **Minimal Overhead**: Initial benchmarks show that enabling detailed metrics
+  incurs a negligible overhead of approximately 0.7% in QPS.
+
+## Efficiency
+
+Valkey 8.0 introduces key embedding in the main dictionary, eliminating separate
+key pointers and significantly reducing memory overhead. This results in a 9-10%
+reduction in overall memory usage for scenarios with 16-byte keys and 8 or 16-byte
+values, along with slight performance improvements.
+
+## Additional Highlights
+
+- **Dual IPv4 and IPv6 Stack Support**: Seamlessly operate in mixed IP environments
+  for enhanced compatibility and flexibility.
+- **Improved Pub/Sub Efficiency**: Lightweight cluster messages streamline
+  communication and reduce overhead for faster, more efficient Pub/Sub operations.
+- **Valkey Over RDMA (Experimental)**: Unlock significant performance improvements
+  with direct memory access between clients and Valkey servers, delivering up to
+  275% increase in throughput and over 50% latency reduction in certain scenarios.
```

Comment:
```suggestion
  275% increase in throughput.
```
My personal take is we aren't materially reducing latency here, so we shouldn't call it out. Latency is a very sensitive topic, and normally the result of queueing. Throughput is pretty easy to measure.